tools/convert/main.go

```diff
@@ -256,6 +255,79 @@ func loadRulesMetadata() *config.Metadata {
 	return m
 }
 
+func ConvertConfig(tmplData *configTemplateData, w io.Writer) {
+	tmpl := template.New("configV2.tmpl")
+	tmpl.Funcs(helpers())
+	tmpl, err := tmpl.ParseFS(filesystem, "templates/configV2.tmpl")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "template error %v\n", err)
+		os.Exit(1)
+	}
+
+	err = tmpl.Execute(w, tmplData)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "template error %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func ConvertHelm(tmplData *configTemplateData, w io.Writer) {
+	// convert config if we have it
+	helmConfigAny, ok := tmplData.Data["config"]
+	if ok {
+		helmConfig, ok := helmConfigAny.(map[string]any)
+		if !ok {
+			panic("config in helm chart is the wrong format!")
+		}
+		convertedConfig := &bytes.Buffer{}
+		// make a copy of this tmplData and overwrite the Data part
+		config := *tmplData
+		config.Data = helmConfig
+		// convert the config into the buffer
+		ConvertConfig(&config, convertedConfig)
+		// read the buffer as YAML
+		decoder := yaml.NewDecoder(convertedConfig)
+		var decodedConfig map[string]any
+		saved := convertedConfig.String()
+		err := decoder.Decode(&decodedConfig)
+		if err != nil {
+			s := err.Error()
+			pat := regexp.MustCompile("yaml: line ([0-9]+):")
+			m := pat.FindStringSubmatch(s)
+			if len(m) > 1 {
+				linenum, _ := strconv.Atoi(m[1])
+				lines := strings.Split(saved, "\n")
+				for i := linenum - 15; i < linenum+5; i++ {
+					if len(lines) > i {
+						fmt.Printf("%d: %s\n", i, lines[i])
+					}
+				}
+			}
+			panic(err)
+		}
+		tmplData.Data["config"] = decodedConfig
+	}
+
+	// now try the rules
+	helmRulesAny, ok := tmplData.Data["rules"]
+	if ok {
+		helmRules, ok := helmRulesAny.(map[string]any)
+		if !ok {
+			panic("config in helm chart is the wrong format!")
+		}
+		rules := convertRulesToNewConfig(helmRules)
```

Comment:
The values.yaml `rules` section may contain a field named `LiveReload ` that is now incompatible with Refinery's stance against extraneous rule fields.  We'll need to remove the entry and move it to the root of the values.yaml.